File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.rb
# @param {Integer} n
# @return {Integer}
def tribonacci(n)
    trib(n, 0, 1, 1)
end

def trib(n, a, b, c)
    return a if n == 0
    return b if n == 1
    return c if n == 2
    trib(n - 1, b, c, a + b + c)
end

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        f = []
        f.append(0)
        f.append(1)
        f.append(1)
        for i in range(3, n + 1):
            s = f[i-1]+f[i-2]+f[i-3]
            f.append(s)
        return f[n]